sqlglot/dialects/dialect.py

```diff
@@ -153,12 +155,52 @@ def _try_load(cls, key: str | Dialects) -> None:
         if isinstance(key, Dialects):
             key = key.value
 
-        # This import will lead to a new dialect being loaded, and hence, registered.
-        # We check that the key is an actual sqlglot module to avoid blindly importing
-        # files. Custom user dialects need to be imported at the top-level package, in
-        # order for them to be registered as soon as possible.
+        # 1. Try standard sqlglot modules first
         if key in DIALECT_MODULE_NAMES:
+            module = importlib.import_module(f"sqlglot.dialects.{key}")
+            # If module was already imported, the class may not be in _classes
+            # Find and register the dialect class from the module
+            if key not in cls._classes:
+                for attr_name in dir(module):
+                    attr = getattr(module, attr_name, None)
+                    if (
+                        isinstance(attr, type)
+                        and issubclass(attr, Dialect)
+                        and attr.__name__.lower() == key
+                    ):
+                        cls._classes[key] = attr
+                        break
+            return
+
+        # 2. Try entry points (for plugins)
+        try:
+            all_eps = entry_points()
+            # Python 3.10+ has select() method, older versions use dict-like access
+            if hasattr(all_eps, "select"):
+                eps = all_eps.select(group="sqlglot.dialects", name=key)
```

Comment:
How does this work? Why are we checking the `sqlglot.dialects` group?

Reply:
We just have to make an agreement how this group should be named and describe this in instructions (I've made README update)
so I chose "sqlglot.dialects" with rely on a project structure, probably there will be more "sqlglot.*" plugin types in future, idk, you can rename this as you wish

Reply:
fixed